src/readii/loaders.py

```diff
@@ -49,11 +54,12 @@ def loadRTSTRUCTSITK(
         A dictionary of mask ROIs from the loaded RTSTRUCT image as a SimpleITK image objects.
         The segmentation label is set to 1.
     """
-
     # Set up segmentation loader
+    logger.debug(f"Making mask using ROI names: {roiNames}")
     makeMask = StructureSetToSegmentation(roi_names=roiNames)
 
     # Read in the base image (CT) and segmentation DICOMs into SITK Images
+    logger.debug(f"Loading RTSTRUCT from directory: {rtstructPath}")
     baseImage = read_dicom_auto(baseImageDirPath)
```

Comment:
`read_dicom_auto` can return any one of `(Scan | PET | StructureSet | Dose | Segmentation | None)`

for `baseImage` I'm assuming it should must return a `Scan` type? 
reason: 

the `baseImage.image` will only work if `baseImage` is of type `Scan`, and we can provide better logging/error handling
```
segMasks = makeMask(
    segImage,
    baseImage.image,
    existing_roi_indices={"background": 0},
    ignore_missing_regex=False,
)
```

if so, should verify this type before proceeding with something like:

```
from imgtools.modules import scan
...
baseImage = read_dicom_auto(baseImageDirPath)
assert isinstance(baseImage, scan.Scan), f"baseImage is not a Scan object. has type {type(baseImage)}"
```

or something

Reply:
Yeah this is a good point. We can actually just update that call to use read_dicom_scan because that's what the function passes it to for both CT and MR. Should retain the assertion though.